examples/performance/iterator-offsets/iterator-offsets.cxx

```diff
@@ -0,0 +1,291 @@
+/*
+ * Testing performance of iterators over the mesh
+ *
+ */
+
+#include <bout.hxx>
+
+#include <chrono>
+#include <iostream>
+#include <iterator>
+#include <vector>
+#include <iomanip>
+
+#include "bout/openmpwrap.hxx"
+#include "bout/region.hxx"
+
+using SteadyClock = std::chrono::time_point<std::chrono::steady_clock>;
+using Duration = std::chrono::duration<double>;
+using namespace std::chrono;
+
+#define ITERATOR_TEST_BLOCK(NAME, ...)                                                   \
+  {                                                                                      \
+    __VA_ARGS__                                                                          \
+    names.push_back(NAME);                                                               \
+    SteadyClock start = steady_clock::now();                                             \
+    for (int repetitionIndex = 0; repetitionIndex < NUM_LOOPS; repetitionIndex++) {      \
+      __VA_ARGS__;                                                                       \
+    }                                                                                    \
+    times.push_back(steady_clock::now() - start);                                        \
+  }
+
+int main(int argc, char **argv) {
+  BoutInitialise(argc, argv);
+  std::vector<std::string> names;
+  std::vector<Duration> times;
+
+  //Get options root
+  Options *globalOptions = Options::getRoot();
+  Options *modelOpts = globalOptions->getSection("performanceIterator");
+  int NUM_LOOPS;
+  OPTION(modelOpts, NUM_LOOPS, 100);
+  bool profileMode, includeHeader;
+  OPTION(modelOpts, profileMode, false);
+  OPTION(modelOpts, includeHeader, false);
+
+  ConditionalOutput time_output{Output::getInstance()};
+  time_output.enable(true);
+
+  const Field3D a{1.0};
+  const Field3D b{2.0};
+
+  Field3D result;
+  result.allocate();
+
+  const int len = mesh->LocalNx*mesh->LocalNy*mesh->LocalNz;
+
+  // Nested loops over block data
+  ITERATOR_TEST_BLOCK(
+    "Nested loop",
+    for(int i=0;i<mesh->LocalNx;++i) {
+      for(int j=mesh->ystart;j<mesh->yend;++j) {
+        for(int k=0;k<mesh->LocalNz;++k) {
+          result(i,j,k) = (a(i,j+1,k) - a(i,j-1,k));
+        }
+      }
+    }
+    );
+
+#ifdef _OPENMP
+  ITERATOR_TEST_BLOCK(
+    "Nested loop (omp)",
+    BOUT_OMP(parallel for)
+    for(int i=0;i<mesh->LocalNx;++i) {
+      for(int j=mesh->ystart;j<mesh->yend;++j) {
+        for(int k=0;k<mesh->LocalNz;++k) {
+          result(i,j,k) = (a(i,j+1,k) - a(i,j-1,k));
+        }
+      }
+    }
+    );
+#endif
+
+  // Range based for DataIterator with indices
+  ITERATOR_TEST_BLOCK(
```

Comment:
I think this (and several below) should be inside the above openmp preprocessor block.